### ISSUE TITLE
masked_select代码未对齐

### ISSUE THREAD

**Issue by BakerMara** [Oneflow-Inc/oneflow#7983]
## Summary

masked_select代码未对齐

## Code to reproduce bug
torch代码可以运行
```python
import torch
x = torch.rand(2, 3, 3)
mask = torch.triu(torch.ones(3, 3), 1).type(torch.ByteTensor)
flat_upper_triange = torch.masked_select(x, mask)
print(flat_upper_triange)
```
oneflow代码
```python
import oneflow as flow
x = flow.rand(2, 3, 3)
mask = flow.triu(flow.ones(3, 3), 1).type(flow.ByteTensor)
flat_upper_triange = flow.masked_select(x, mask)
print(flat_upper_triange)
```
> 首先会提示'oneflow._oneflow_internal.Tensor' object has no attribute 'type'

修改为以下代码后
```python
import oneflow as flow
x = flow.rand(2, 3, 3)
mask = flow.triu(flow.ones(3, 3), 1)
flat_upper_triange = flow.masked_select(x, mask)
print(flat_upper_triange)
```
> 报错信息：The dim of masked_select module's inputs can not match, please check!

flow.masked_select源码
```python
def masked_select_op(input, mask):
    assert len(input.shape) == len(
        mask.shape
    ), f"The dim of masked_select module's inputs can not match, please check!"
```
此处限定了维度必须相同，导致无法进行广播

修改为以下代码后可运行
```python
import oneflow as flow
x = flow.rand(2, 3, 3)
mask = flow.triu(flow.ones(3, 3), 1)
flat_upper_triange = flow.masked_select(x, mask.unsqueeze(0))
print(flat_upper_triange)
```

但是采取同样的方法在graph代码中修改则会提示以下错误：
>building graph got error: <class 'ValueError'> A lazy tensor can not be applied to argwhere.
